models/creditCardModel.js

```diff
@@ -1,51 +1,124 @@
-// models/creditCardModel.js
+const validator = require('validator'); // For validating inputs like email, card number, etc.
+const sanitizeHtml = require('sanitize-html'); // To sanitize any HTML input
+var luhn = require("luhn");
 
-const createCreditCard = (firstName, lastName, contact, Address, cardHolderName, cardNo, expiryDate, cvvCode, callback) => {
-    // Check if the card already exists
-    const querymatch = 'SELECT * FROM tbl_cards WHERE cardNo = ?';
-    global.db.query(querymatch, [cardNo], (err, results) => {
+const createCreditCard = (firstName, lastName, contact, Address, Email, cardHolderName, cardNo, expiryDate, cvvCode, callback) => {
+  // Step 1: Input sanitization
+  
+  // Sanitize strings to prevent any HTML or script injections
+  firstName = sanitizeHtml(firstName);
+  lastName = sanitizeHtml(lastName);
+  contact = sanitizeHtml(contact);
+  Address = sanitizeHtml(Address);
+  cardHolderName = sanitizeHtml(cardHolderName);
+  
+  // Sanitize and validate email
+  Email = sanitizeHtml(Email);
+  if (!validator.isEmail(Email)) {
+    return callback({ message: 'Invalid email format.' }, null);
+  }
+
+  // Validate and sanitize card number and CVV code (ensure they are numeric and valid)
+  cardNo = cardNo.replace(/\D/g, ''); // Remove any non-digit characters
+  cvvCode = cvvCode.replace(/\D/g, ''); // Remove any non-digit characters
+
+  // Step 2: Input validation
+
+  // Validate the card number using Luhn algorithm (for example, using validator library or custom logic)
+  if (!luhn.validate(cardNo)) {
+    return callback({ message: 'Provide correct card no.' }, null);
+  }
+
+
+  if (!validator.isNumeric(cardNo)) {
+    return callback({ message: 'Card number must only contain numeric characters.' }, null);
+  }
+
+  // Validate expiry date format (MM/YY)
+  if (!validator.matches(expiryDate, /^(0[1-9]|1[0-2])\/\d{2}$/)) {
```

Comment:
Use Validator library instead of regex

Reply:
I have used the NPM library now to resolve this